meteor/client/lib/triggers/TriggersHandler.tsx

```diff
@@ -0,0 +1,424 @@
+			}
+		})
+	)
+	if (Tracker.currentComputation) {
+		Tracker.currentComputation.onStop(() => {
+			computation.stop()
+		})
+	}
+	return computation
+}
+
+/**
+ * Note: there can only be a single TriggersHandler in the node tree.
+ *
+ * @param {IProps} props
+ * @return {*}
+ */
+export const TriggersHandler: React.FC<IProps> = function TriggersHandler(
+	props: IProps
+): React.ReactElement<any, any> | null {
+	const [initialized, setInitialized] = useState(props.sorensen ? true : false)
+	const { t } = useTranslation()
+	const localSorensen = props.sorensen || Sorensen
+
+	function bindHotkey(id: TriggeredActionId, keys: string, up: boolean, action: HotkeyTriggerListener) {
+		try {
+			localSorensen.bind(keys, action, {
+				up,
+				exclusive: true,
+				ordered: 'modifiersFirst',
+				global: props.global ?? false,
+				tag: id,
+			})
+		} catch (e) {
+			console.error(e)
+		}
+	}
+
+	function unbindHotkey(keys: string, listener: (e: KeyboardEvent) => void) {
+		localSorensen.unbind(keys, listener)
+	}
+
+	useEffect(() => {
+		// if we're using a local instance of Sorensen
+		if (!props.sorensen) {
+			localSorensen
+				.init()
+				.then(() => {
+					setInitialized(true)
+				})
+				.catch(console.error)
+		}
+
+		return () => {
+			// do not destroy, if we're using a provided instance of Sorensen
+			if (!props.sorensen) {
+				localSorensen.destroy().catch(console.error)
+			}
+		}
+	}, []) // run once
+
+	useEffect(() => {
+		if (rundownPlaylistContext.get() !== null) {
+			throw new Error('There can only be a single instance of TriggersHandler in the Node tree.')
+		}
+
+		return () => {
+			setRundownPlaylistContext(null)
+		}
+	}, [])
+
+	function poisonHotkeys() {
+		console.log('Poisoned')
+		localSorensen.poison() // cancels all pressed keys, poisons all chords, no hotkey trigger will execute
+	}
+
+	useEffect(() => {
+		if (initialized) {
+			localSorensen.bind('Escape', poisonHotkeys, {
+				exclusive: false,
+				global: true,
+			})
+		}
+
+		return () => {
+			localSorensen.unbind('Escape', poisonHotkeys)
+		}
+	}, [initialized]) // run once once Sorensen is initialized
+
+	useEffect(() => {
+		Tracker.nonreactive(() => {
+			const playlist = RundownPlaylists.findOne(props.rundownPlaylistId, {
+				fields: {
+					_id: 1,
+					name: 1,
+					activationId: 1,
+					nextPartInstanceId: 1,
+					currentPartInstanceId: 1,
+				},
+			})
+			if (playlist) {
+				setRundownPlaylistContext({
+					rundownPlaylist: playlist,
+					currentRundownId: props.currentRundownId,
+					currentPartId: props.currentPartId,
+					nextPartId: props.nextPartId,
+					currentSegmentPartIds: props.currentSegmentPartIds,
+					nextSegmentPartIds: props.nextSegmentPartIds,
+				})
+			}
+		})
+	}, [
+		props.rundownPlaylistId,
+		props.currentRundownId,
+		props.currentPartId,
+		props.currentSegmentPartIds,
+		props.nextPartId,
+		props.nextSegmentPartIds,
+	])
+
+	const triggerSubReady = useSubscription(PubSub.triggeredActions, {
+		$or: [
```

Comment:
Mongodb recommends to do `$in: [null, props.showStyleBaseId]` instead (I suspect it might slightly improve db-performance), see https://docs.mongodb.com/manual/reference/operator/query/or/#-or-versus--in

Reply:
Yeah, that's a good note.